src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -1865,6 +1938,43 @@ void M()
             Assert.Equal("delegate*<ref System.Void> ptr5", model.GetDeclaredSymbol(decls[4]).ToTestDisplayString());
         }
 
+        [Fact]
+        public void PublicApi_NonApplicationCorLibrary()
+        {
+            var otherCorLib = CreateEmptyCompilation(@"
+namespace System
+{
+    public class Object { }
+    public abstract class ValueType { }
+    public struct Void { }
+    public class String { }
+    namespace Runtime.CompilerServices
+    {
+        internal class CallConvTest {}
+        public static class RuntimeFeature
+        {
+            public const string UnmanagedSignatureCallingConvention = nameof(UnmanagedSignatureCallingConvention);
+        }
+    }
+}
+", options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularPreview);
+
+            var mainComp = CreateCompilation("", references: new[] { otherCorLib.ToMetadataReference() });
+            var returnType = mainComp.GetSpecialType(SpecialType.System_String).GetPublicSymbol();
+            var testConvention = otherCorLib.GetTypeByMetadataName("System.Runtime.CompilerServices.CallConvTest");
+            Assert.NotNull(testConvention);
+            Assert.True((object)testConvention!.ContainingAssembly.CorLibrary != mainComp.Assembly.CorLibrary);
```

Comment:
>Assert.True((object)testConvention!.ContainingAssembly.CorLibrary != mainComp.Assembly.CorLibrary); [](start = 12, length = 99)

This feels confusing, why do we even giving ```otherCorLib``` as a reference to ```mainComp```? Is this creating some kind of ambiguity between the references? I think it would be better to avoid creating a possibility of other kind of errors to some how interfere with the scenario that we are trying to test, if I understand that scenario correctly. I suggest removing the ```otherCorLib``` from the set of references given to ```mainComp```. Perhaps I am misinterpreting the intent and creation of ambiguity among references is intentional. Then, I agree, it is worth keeping this test too. However, then I would suggest verifying that, if ```otherCorLib``` reference comes first, then ```CreateFunctionPointerTypeSymbol``` succeeds.